core/backend/src/SqliteStatement.ts

```diff
@@ -158,57 +165,117 @@ export class SqliteStatement implements IterableIterator<any>, IDisposable {
       throw new IModelError(stat, "Error in bindValue");
   }
 
-  private checkBind(stat: DbResult) {
-    if (stat !== DbResult.BE_SQLITE_OK)
-      throw new IModelError(stat, "SQLite Bind error");
-  }
   /** Bind an integer parameter
-   *  @param parameter Index (1-based) or name of the parameter (including the initial ':', '@' or '$')
-   *  @param val integer to bind.
+   * @param parameter Index (1-based) or name of the parameter (including the initial ':', '@' or '$')
+   * @param val integer to bind.
    */
   public bindInteger(parameter: BindParameter, val: number) {
-    this.checkBind(this.stmt.bindInteger(parameter, val));
+    checkBind(this.stmt.bindInteger(parameter, val));
+  }
+  /** Bind an integer parameter if it is defined. Otherwise do nothing.
+   * @param parameter Index (1-based) or name of the parameter (including the initial ':', '@' or '$')
+   * @param val integer to bind.
+   */
+  public maybeBindInteger(parameter: BindParameter, val?: number) {
+    if (val)
```

Comment:
What if `val` is `0`?

Reply:
yea, i noticed that too

Reply:
i'm moving my tests for all of this into this repo. I'll do that next.